packages/instant/src/util/heap.ts

```diff
@@ -0,0 +1,85 @@
+import { ObjectMap } from '@0x/types';
+
+import { HEAP_ANALYTICS_DEVELOPMENT_APP_ID } from '../constants';
+
+export interface HeapAnalytics {
+    loaded: boolean;
+    identify(id: string, idType: string): void;
+    track(eventName: string, eventProperties?: ObjectMap<string | number>): void;
+    resetIdentity(): void;
+    addUserProperties(properties: ObjectMap<string | number>): void;
+    addEventProperties(properties: ObjectMap<string | number>): void;
+    removeEventProperty(property: string): void;
+    clearEventProperties(): void;
+}
+interface ModifiedWindow {
+    heap?: HeapAnalytics;
+    zeroExInstantLoadedHeap?: boolean;
+}
+const getWindow = (): ModifiedWindow => {
+    return window as ModifiedWindow;
+};
+
+const setupZeroExInstantHeap = () => {
+    const curWindow = getWindow();
+    // Set property to specify that this is zeroEx's heap
+    curWindow.zeroExInstantLoadedHeap = true;
+
+    // Typescript-compatible version of https://docs.heapanalytics.com/docs/installation
+    /* tslint:disable */
+    ((window as any).heap = (window as any).heap || []),
+        ((window as any).heap.load = function(e: any, t: any) {
+            ((window as any).heap.appid = e), ((window as any).heap.config = t = t || {});
+            var r = t.forceSSL || 'https:' === (document.location as Location).protocol,
+                a = document.createElement('script');
+            (a.type = 'text/javascript'),
+                (a.async = !0),
+                (a.src = (r ? 'https:' : 'http:') + '//cdn.heapanalytics.com/js/heap-' + e + '.js');
+            var n = document.getElementsByTagName('script')[0];
+            (n.parentNode as Node).insertBefore(a, n);
+            for (
+                var o = function(e: any) {
+                        return function() {
+                            (window as any).heap.push([e].concat(Array.prototype.slice.call(arguments, 0)));
+                        };
+                    },
+                    p = [
+                        'addEventProperties',
+                        'addUserProperties',
+                        'clearEventProperties',
+                        'identify',
+                        'resetIdentity',
+                        'removeEventProperty',
+                        'setEventProperties',
+                        'track',
+                        'unsetEventProperty',
+                    ],
+                    c = 0;
+                c < p.length;
+                c++
+            )
+                (window as any).heap[p[c]] = o(p[c]);
+        });
+    // TODO: use production heap id once environment utils merged
+    (window as any).heap.load(HEAP_ANALYTICS_DEVELOPMENT_APP_ID);
+    /* tslint:enable */
+
+    return curWindow.heap as HeapAnalytics;
```

Comment:
This might now work... did you see the heap GOTCHA in website?

``
    /**
     * Heap is not available as a UMD module, and additionally has the strange property of replacing itself with
     * a different object once it's loaded.
     * Instead of having an await call before every analytics use, we opt to have the awaiting logic in here by
     * guarding every API call with the guard below.
     */
    private async _heapLoadedGuardAsync(callback: () => void): Promise<void> {
        if (this._heap.loaded) {
            callback();
            return undefined;
        }
        await utils.onPageLoadPromise;
        // HACK: Reset heap to loaded heap
        this._heap = (window as any).heap;
        callback();
    }
```

Reply:
Never mind this works.